### PR TITLE
[REEF-437]:Fix DefaultRemoteManagerImplementation.java to enable

### PR DESCRIPTION
**Pull request by beysims**
TcpPortProvider configuration

This addressed the issue by adding a new constructor that takes
TcpPortProvider

JIRA: [REEF-437](https://issues.apache.org/jira/browse/REEF-437)

This closes #

### CODE REVIEW

File: lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java [hunk 1]
```diff
@@ -47,7 +49,7 @@
 
   private static final Logger LOG = Logger.getLogger(HandlerContainer.class.getName());
 
-  private static final AtomicInteger COUNTER = new AtomicInteger(0);
+  private static final AtomicInteger counter = new AtomicInteger(0);
```
Comment (markusweimer) [inline lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java]: why this change? it seems orthogonal to the needs of the JIRA. Also, I believe this should violate the checkstyle, but @tcNickolas knows best.

File: lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java [hunk 2]
```diff
@@ -107,34 +109,51 @@
...
-        hostAddress, listeningPort, this.reRecvStage, this.reRecvStage, numberOfTries, retryTimeout);
-
-    this.handlerContainer.setTransport(this.transport);
-
-    this.myIdentifier = new SocketRemoteIdentifier(
-        (InetSocketAddress) this.transport.getLocalAddress());
-
-    this.reSendStage = new RemoteSenderStage(codec, this.transport, 10);
+    @Inject
+    public <T> DefaultRemoteManagerImplementation(
```
Comment (markusweimer) [inline lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java]: This constructor is added to be used by Tang only, right? If so, can we make it `private`?

File: lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java [hunk 3]
```diff
@@ -107,34 +109,51 @@
...
 
-    LOG.log(Level.FINEST, "RemoteManager {0} instantiated id {1} counter {2} listening on {3}:{4}. Binding address provided by {5}",
-        new Object[]{this.name, this.myIdentifier, COUNTER.incrementAndGet(),
-            this.transport.getLocalAddress().toString(),
-            this.transport.getListeningPort(), localAddressProvider}
-    );
-  }
-
-  /**
+    /**
```
Comment (markusweimer) [inline lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java]: Pure formatting change, please remove.

**Comment by markusweimer** [comment]
@beysims I've done a pass.

File: lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java (hunk 1 above)
Comment (tcNickolas) [inline lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java]: Yes, this will violate ConstantName check (static final field => must be ALL_CAPS). Currently this check is warning-level, not error-level, because we still have some misnamed constants which are public (so can't be renamed easily, have to do deprecation process). I will go through such violations and add exceptions for them so that we can enforce all checks in Naming category and stop introducing new violations

Comment (beysims) [inline lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java]: I do not remember changing this, but fixed.

File: lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java (hunk 2 above)
Comment (beysims) [inline lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java]: fixed

File: lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java (hunk 3 above)
Comment (beysims) [inline lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java]: fixed

**Comment by beysims** [comment]
@markusweimer, I have addressed your review comments.

**Comment by markusweimer** [comment]
I'll merge this.
